dbms/src/Storages/Page/V3/Universal/UniversalPageStorageService.cpp

```diff
@@ -12,33 +12,201 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <Common/Exception.h>
 #include <Interpreters/Context.h>
+#include <Poco/File.h>
+#include <Poco/Path.h>
+#include <Storages/BackgroundProcessingPool.h>
+#include <Storages/DeltaMerge/Remote/DataStore/DataStore.h>
+#include <Storages/Page/V3/Universal/UniversalPageStorage.h>
 #include <Storages/Page/V3/Universal/UniversalPageStorageService.h>
+#include <Storages/S3/S3Common.h>
+#include <Storages/S3/S3Filename.h>
+#include <Storages/Transaction/KVStore.h>
+#include <Storages/Transaction/TMTContext.h>
+#include <Storages/Transaction/Types.h>
+#include <aws/s3/S3Client.h>
+
+#include <ext/scope_guard.h>
 
 namespace DB
 {
+
+UniversalPageStorageService::UniversalPageStorageService(Context & global_context_)
+    : global_context(global_context_)
+    , uni_page_storage(nullptr)
+    , log(Logger::get())
+{
+    checkpoint_pool = std::make_unique<BackgroundProcessingPool>(1, "ps-checkpoint");
```

Comment:
(Not important) Is it possible that we only create `checkpoint_pool` if `s3factory.isEnabled() == true`?